Expensify/react-native-live-markdown
# Add selection variables to web implementation

### Details

This PR:
- adds `selection` prop support on web
- fixes `selectionStart` and `selectionEnd` variables behavior inside MarkdownTextInput ref on web
- fixes selection detection on web 

### Related Issues

[GH_LINK](https://github.com/Expensify/App/issues/36199)

### Manual Tests

1. Start writing in MarkdownTextInput on web
2. Verify if cursor position is correct when writing casually
3. Verify if text replacement works properly. Select some text and start writing 
4. Verify if after pasting text or emojis cursor is being set in correct position
5. Add `onSelectionChange` handler to MarkdownTextInput and verify if event returns correct selection start value
6. Add `ref` to MarkdownTextInput and verify if `ref.selectionStart` and `ref.selectionEnd` returns correct  selection end value

### Linked PRs
